Convex hull list sample

Rachael-E opened the pull request:
Implementation of Convex Hull List Sample, ready for code review. Image and README.md file included.

Rachael-E committed:
Initiating Convex Hull List Sample

Setting up file structure for sample, initialising README and .java file

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Setting up stackpane, mapview, buttons and checkbox.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Hard coded polygons for displaying on map.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Build up graphics overlay for each polygon and for convex hull. Addede functionality to return convex hull of the first polygon.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Set up clear button on action to clear convex hull graphic.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Setting up Geometry Engine for Convex Hull

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Refactored code to use Polygon rather than Polygon Builder

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Refactoring

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Refactor code to make local variables more meaningful, and add comments to code.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Added a border to controls VBox

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Adjusting label extents and wrapping text

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Ensured correct placement of label and buttons in VBox, re-ordered import packages. Set up zoom level for map when loaded.

Rachael-E committed:
Updates to Convex Hull List

Added an image and README file to implementation, refactor of code to allow correct graphic overlay order.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/README.md

Edits to README.md

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/README.md

Close code tag in README.md

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

edits to comments to ensure consistency in style

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+      });
+
+      // create label
+      Label informationLabel = new Label(
```
.NET samples shows instructions in their samples. Java chose not to do this because it clutters the UI and we already have the readme and Info tab in the sample viewer. So you can remove this label.

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+
+      // create a control panel
+      VBox controlsVBox = new VBox(6);
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(255, 255, 255, 1)"),
```
the background should be semi-transparent black like in other samples

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+      });
+
+      // disable clear button from starting application (when nothing to clear)
+      resetButton.setDisable(true);
```
would be good to call this `clearButton` with text "Clear" like in the other convex hull sample

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+        Boolean unionBool = checkBox.isSelected();
+
+        // add the geometries of the two polygon graphics to a list of geometries
+        List<Geometry> allPolygonGeometries = new ArrayList<>();
```
Could use `Arrays.asList(polygonGraphic1.getGeometry(), polygonGraphic2.getGeometry())`

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+        // reset the convex hull graphics overlay
+        convexHullGraphicsOverlay.getGraphics().clear();
+
+        Boolean unionBool = checkBox.isSelected();
```
don't need a variable here since it's only used once

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+      graphicsOverlay.getGraphics().add(polygonGraphic2);
+
+      // create a button to create and show the convex hull
+      Button convexHullButton = new Button("Convex Hull");
```
"Create Convex Hull" for consistency with the other sample

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+      mapView.setMap(map);
+
+      // create a graphics overlay for the two input polygons
+      GraphicsOverlay graphicsOverlay = new GraphicsOverlay();
```
maybe `polygonsOverlay`?

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java:
```diff
@@ -0,0 +1,255 @@
+        allPolygonGeometries.add(polygonGraphic2.getGeometry());
+
+        // retrieve the returned result from the convex hull operation.
+        // when unionBool = true there will be one returned convex hull geometry
```
if unioned, one geometry is returned, otherwise one convex hull geometry is returned per input geometry

tschie reviewed (changes requested):
Looks good. Just a few recommendations

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/README.md:
```diff
@@ -0,0 +1,37 @@
+
+<h2>How it works</h2>
+
+<p>The static method, <code>GeometryEngine.ConvexHull</code>, is called by passing in a list of polygon geometries and a boolean to union the result if desired. The two input polygons geometries are hard-coded as part of the sample initialization. </p>
```
You can remove this. It's redundant with the list below. The Java SDK deviates slightly from others by always just giving a list in this section.

tschie commented on src/main/java/com/esri/samples/geometry/convex_hull_list/README.md:
```diff
@@ -0,0 +1,37 @@
+<h1>Convex Hull List</h1>
+
+<p>Generate convex hull polygon(s) from multiple input geometries. As a visual analogy, consider a set of points as nails in a board. The convex hull of the points would be like a rubber band stretched around the outermost nails.</p>
```
The visual analogy part can go in a separate `p` tag. The first p tag should always be pretty short for the sample viewer short description.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Change background colour of control panel to transparent black

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

removed label from sample

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Rename "reset" button to "clear" button

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Change convex hull button text, and move checkBox.isSelected directly into Geometry Enginer paramater

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Remove unused import statements, gave the polygon graphics overlay a meaningful name, refactored ArrayList to Arrays.asList.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Reduced the size of the VBox.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/README.md

Removed intro from 'How it Works' section following review.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullListSample.java

Removed unused import

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/ConvexHullList.png

Updated image following review chanes.

Rachael-E committed:
Update src/main/java/com/esri/samples/geometry/convex_hull_list/README.md

Update text to reflect edited button names

Rachael-E committed:
Update ConvexHullListSample.java

Check box should remain checked when convex hull is clicked for unioning result - feedback addressed
